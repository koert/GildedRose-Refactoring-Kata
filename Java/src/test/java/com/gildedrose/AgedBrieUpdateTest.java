package com.gildedrose;

import static org.junit.jupiter.api.Assertions.*;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

/**
 * @author Koert Zeilstra
 */
class AgedBrieUpdateTest {

  public static final String NAME = "Aged Brie";
  public static final Integer[] SELL_IN = {-1, 0, 4, 5, 6, 9, 10, 11};
  public static final Integer[] QUALITY = {0, 10, 50};

  AgedBrieUpdate update = new AgedBrieUpdate();

  @DisplayName("update")
  @Test
  void update() {
    CombinationApprovals.verifyAllCombinations(this::doUpdateQuality, SELL_IN, QUALITY);
  }

  String doUpdateQuality(int sellIn, int quality) {
    Item item = new Item(NAME, sellIn, quality);    
    update.update(item);
    return item.toString();
  }

}