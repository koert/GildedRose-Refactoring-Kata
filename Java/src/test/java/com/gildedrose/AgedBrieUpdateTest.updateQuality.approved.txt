[-1, 0] => Aged Brie, -2, 2 
[-1, 10] => Aged Brie, -2, 12 
[-1, 50] => Aged Brie, -2, 50 
[0, 0] => Aged Brie, -1, 2 
[0, 10] => Aged Brie, -1, 12 
[0, 50] => Aged Brie, -1, 50 
[4, 0] => Aged Brie, 3, 1 
[4, 10] => Aged Brie, 3, 11 
[4, 50] => Aged Brie, 3, 50 
[5, 0] => Aged Brie, 4, 1 
[5, 10] => Aged Brie, 4, 11 
[5, 50] => Aged Brie, 4, 50 
[6, 0] => Aged Brie, 5, 1 
[6, 10] => Aged Brie, 5, 11 
[6, 50] => Aged Brie, 5, 50 
[9, 0] => Aged Brie, 8, 1 
[9, 10] => Aged Brie, 8, 11 
[9, 50] => Aged Brie, 8, 50 
[10, 0] => Aged Brie, 9, 1 
[10, 10] => Aged Brie, 9, 11 
[10, 50] => Aged Brie, 9, 50 
[11, 0] => Aged Brie, 10, 1 
[11, 10] => Aged Brie, 10, 11 
[11, 50] => Aged Brie, 10, 50 
